client/components/data/query-top-posts/index.jsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { isUndefined } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingTopPosts } from 'state/stats/top-posts/selectors';
+import { requestTopPosts } from 'state/stats/top-posts/actions';
+
+class QueryTopPosts extends Component {
+	static defaultProps = {
+		requestTopPosts: () => {},
+		period: 'day',
+		num: 1,
+	};
+
+	static propTypes = {
+		siteId: PropTypes.number,
+		date: PropTypes.string,
+		period: PropTypes.string,
+		num: PropTypes.number,
+		requestingTopPosts: PropTypes.bool,
+		requestTopPosts: PropTypes.func,
+	};
```

Comment:
For now, the max number is the default provided by the API: `10`. I was trying to keep the parameters to a minimum for a first version, but this is an important one to have from the beginning, I will add it.